server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -721,6 +721,10 @@ private Pair<List<EventJoinVO>, Integer> searchForEventsInternal(ListEventsCmd c
         ListProjectResourcesCriteria listProjectResourcesCriteria = domainIdRecursiveListProject.third();
 
         Filter searchFilter = new Filter(EventJoinVO.class, "createDate", false, cmd.getStartIndex(), cmd.getPageSizeVal());
+        // additional order by since createdDate does not have milliseconds
+        // and two events, created within one second can be incorrectly ordered (for example VM.CREATE Completed before Scheduled)
```

Comment:
yes, a known issue in usage. good fix. I would like this added to javadoc somewhere. (maybe a separate issue, not in this PR).